File: Tests/OrbeEditorCoreTests/Fixtures/samples/sample.go
package main

import "fmt"

// LineIndex holds line start offsets.
type LineIndex struct {
	starts []int
}

func NewLineIndex(text string) *LineIndex {
	starts := []int{0}
	for i, r := range text {
		if r == '\n' {
			starts = append(starts, i+1)
		}
	}
	return &LineIndex{starts: starts}
}

func main() {
	fmt.Println(len(NewLineIndex("a\nb").starts))
}

File: Tests/OrbeEditorCoreTests/Fixtures/samples/sample.swift
import Foundation

/// 行頭オフセットの索引。
public struct LineIndex: Equatable {
  private var starts: [Int]

  public init(text: String) {
    starts = [0]
    var offset = 0
    for unit in text.utf16 {
      offset += 1
      if unit == 0x0A { starts.append(offset) }
    }
  }

  public func point(at offset: Int) -> (row: Int, column: Int) {
    guard let row = starts.lastIndex(where: { $0 <= offset }) else { return (0, 0) }
    return (row, offset - starts[row])
  }
}
